text/0000-deprecation.md

```diff
@@ -0,0 +1,158 @@
+- Feature Name: A plan for deprecating APIs within Rust
+- Start Date: 2015-06-03
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+
+There has been an ongoing [discussion on 
+internals](https://internals.rust-lang.org/t/thoughts-on-aggressive-deprecation-in-libstd/2176/55) 
+about how we are going to evolve the standard library. This RFC tries 
+to condense the consensus.
+
+As a starting point, the current deprecation feature allows a developer
+to annotate their API items with `#[deprecated(since="1.1.0")]` and
+have suitable warnings shown if the feature is used.
+
+# Motivation
+
+We want to:
+
+1. evolve the `std` API, including making items unavailable with new 
+versions
+2. with minimal -- next to no -- breakage
+3. be able to plug security/safety holes 
+4. avoid confusing users
+5. stay backwards-compatible so people can continue to use dependencies 
+written for older versions (except where point 3. forbids this)
```

Comment:
I'm also somewhat skeptical that we'd outright *remove* an API because it's unsafe or insecure, for example `gets` is still in the C standard library, it's just plastered with endless warnings to not use it.